### PR TITLE
[account] LoadAccount() to use address.Address

### LINKED ISSUE #2771: [account] unify API to use address.Address
**What would you like to be added**:
1. change
```
func LoadAccount(sr protocol.StateReader, addrHash hash.Hash160) (*state.Account, error)
``` 
to
```
func LoadAccount(sr protocol.StateReader, addr address.Address) (*state.Account, error)
```
2. rename the original `func LoadAccount()` to `func LoadAccountByHash160()` so `evmstatedbadaptor.go` can continue to use it

**Why is this needed**:
1. be consistent with `StoreAccount` to use `address.Address` as input
2. lots of code like
```
caller, err := accountutil.LoadAccount(sm, hash.BytesToHash160(caller.Bytes()))
```
can reduce to 
```
caller, err := accountutil.LoadAccount(sm, caller))
```

### PR DESCRIPTION
**Pull request by ashishsnigam**
Fixed #2771 
1. Function LoadAccount signature changed and updated calling
2. Original LoadAccount function restored as LoadAccountByHash160

### CODE REVIEW

**Comment by ashishsnigam** [comment]
@dustinxie @CoderZhi Please review this PR

File: action/protocol/account/util/util.go [hunk 1]
```diff
@@ -54,7 +54,21 @@ func LoadOrCreateAccount(sm protocol.StateManager, encodedAddr string) (*state.A
 }
 
 // LoadAccount loads an account state
-func LoadAccount(sr protocol.StateReader, addrHash hash.Hash160) (*state.Account, error) {
+func LoadAccount(sr protocol.StateReader, addr address.Address) (*state.Account, error) {
+	var account state.Account
+	addrHash := hash.BytesToHash160(addr.Bytes())
```
Comment (dustinxie) [inline action/protocol/account/util/util.go:59]: `return LoadAccountByHash160(addrHash)`

Comment (ashishsnigam) [reply action/protocol/account/util/util.go:58]: Ahh..  Done

Comment (ashishsnigam) [reply action/protocol/account/util/util.go:58]: @dustinxie Review comments incorporated

**Review by dustinxie — APPROVED** [review]

File: action/protocol/account/util/util.go (hunk 1 above)
Comment (dustinxie) [reply action/protocol/account/util/util.go]: thx, would you like to pick up https://github.com/iotexproject/iotex-core/issues/2775 as well?

**Review by raullenchai — APPROVED** [review]

**Pull request merged**